### PR TITLE
[OSPP] The proposal for Lifelong Learning O&M

### LINKED ISSUE #306: [Feature] Sedna Edge-Cloud Collaboration Lifelong Learning O&M
**What would you like to be added/modified**:
Based on the existing basic core capabilities of Sedna edge-cloud collaborative lifelong learning, try to complete the following tasks:
1. Using the K8s Dashboard or others to integrate the management and status monitoring of lifelong learning system services, 
lifelong learning task management and status monitoring, and lifelong learning knowledge base management and status monitoring.
2. Using Prometheus to collect and visualize the dynamic metrics of lifelong learning.
3. Using the open-source log management tool to manage and retrieve application logs in a unified manner.

**Other information**:
[Sedna lifelonglearning](https://github.com/kubeedge/sedna/blob/main/docs/proposals/lifelong-learning.md)
[Sedna guide](https://sedna.readthedocs.io/en/latest/index/guide.html)
[How to contribute Sedna](https://sedna.readthedocs.io/en/latest/contributing/prepare-environment.html)

### PR DESCRIPTION
**Pull request by wjf222**
**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
This is the proposal for Lifelong Learning O&M
**Which issue(s) this PR fixes**:

Fixes #306

### CODE REVIEW

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+## proposal
+We proposal using grafana,loki and prometheus to display the mertic,log and statues about the lifelong learnig job. The Prometheus is used to collect the mertic and status and the loki is used to collect logs.
+### use cases
+
+- Users can search log by specifying pod and keywords in the log control panel
+- User can view the history state or the mertic of any component in a sequence diagram
+- Users can view the status of any CRD (Model, DataSet, lifelongjob) by specifying the name
+## design details
+### Architecture
+![](./images/lifelong-learning-ops-architecture.png)
```
Comment (MooreZheng) [inline docs/proposals/Lifelong-learning-Ops-ui.md:29]: How does this proposal work with Sedna, especially the interaction between the proposed modules and sedna modules?

Comment (wjf222) [reply docs/proposals/Lifelong-learning-Ops-ui.md:29]: > How does this proposal work with Sedna, especially the interaction between the proposed modules and sedna modules?

I will add Sedna to the structure diagram.

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
+- [operation and maintenance UI development](#operation-and-maintenance-UI-development)
```
Comment (JimmyYang20) [inline docs/proposals/Lifelong-learning-Ops-ui.md:1]: First letter capitalization
`Operation`

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    - [Design Details](#design-details)
+        - [Loki](#loki)
+        - [Prometheus](#prometheus)
+        - [Grafana](#grafna)
+    - [Install](#install)
+
+# operation and maintenance UI development
+## Moaivation
+At present, Users can only access information about lifelong learning through the command line, including logs, status, and metrics.It is inconvenent and unfriendly for users. This proposal will provide a UI for metrics, log collection, status monitoring and management based on grafana. SThis allows the user to get the information by the graphical interface
+### goals
```
Comment (JimmyYang20) [inline docs/proposals/Lifelong-learning-Ops-ui.md:15]: `SThis` -> `This`

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+- Supports metrics collection and visualization in lifelong learning
+- Support unified management and search of application logs
+- support the management and status monitoring of lifelong learning in dashboard
```
Comment (JimmyYang20) [inline docs/proposals/Lifelong-learning-Ops-ui.md:19]: Please check the full text, paying attention to the word case and punctuation.

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+2. Dataset status
+
+| **Metric**       | **Type** | **Range**                           |
+|------------------|----------|-------------------------------------|
+| NumberOfSamples  | gauge    | The number of samples               |
+| StageTrainNumber | gauge    | The number of samples used by train |
+| StageEvalNumber  | gauge    | The number of samples used by eval  |
+
+3. Model
```
Comment (JimmyYang20) [inline docs/proposals/Lifelong-learning-Ops-ui.md:62]: Lack of algorithm-related metrics, such as metrics to evaluate the overall performance of the lifelong learning paradigm.

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+|                            | Daemonset                 | SideCar                                   |
+|----------------------------|---------------------------|-------------------------------------------|
+| Source                     | Sysout+ part file         | file                                      |
+| Log classification storage | mapped by container/path  | Pod can be separately                     |
+| Multi-tenant isolation     | isolated by configuration | isolated by Pod                           |
+| Resource occupancy         | low                       | high                                      |
+| Customizability            | Low                       | High, each Pod is configured individually |
+| Applicable scene           | Applicable scene          | Large, mixed cluster                      |
+
+### Prometheus
```
Comment (JimmyYang20) [inline docs/proposals/Lifelong-learning-Ops-ui.md:43]: Lack of network solutions for edge nodes.

File: docs/proposals/Lifelong-learning-Ops-ui.md (hunk 1 above)
Comment (MooreZheng) [reply docs/proposals/Lifelong-learning-Ops-ui.md:29]: At present, Sedna modules and the interactions with the proposal are still missing

File: docs/proposals/Lifelong-learning-Ops-ui.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+### goals
+
+- Supports metrics collection and visualization in lifelong learning
+- Support unified management and search of application logs
+- support the management and status monitoring of lifelong learning in dashboard
+## proposal
+We proposal using grafana,loki and prometheus to display the mertic,log and statues about the lifelong learnig job. The Prometheus is used to collect the mertic and status and the loki is used to collect logs.
+### use cases
+
+- Users can search log by specifying pod and keywords in the log control panel
```
Comment (back1860) [inline docs/proposals/Lifelong-learning-Ops-ui.md:24]: 是不是应该支持用户通过终身任务的名称或者ID收缩日志、查看状态？

File: docs/proposals/Lifelong-learning-Ops-ui.md (hunk 1 above)
Comment (back1860) [inline docs/proposals/Lifelong-learning-Ops-ui.md:29]: 重点考虑如下问题：
1. 边缘节点往往在防火墙之后，这种情况下如何在边云之间传输数据；
2. 边缘节点断链的情况下，本地是否缓存数据？

Comment (wjf222) [reply docs/proposals/Lifelong-learning-Ops-ui.md:29]: I have modified the architecture diagram

**Comment by wjf222** [comment]
I will take into account network issues between nodes.

**Comment by luosiqi** [comment]
lgtm

**Comment by wjf222** [comment]
@JimmyYang20 I have updated the instructions for appending algorithm metrics.

**Pull request merged**